lib/ansible/modules/cloud/google/gcspanner.py

```diff
@@ -0,0 +1,265 @@
+    HAS_PYTHON26 = True
+except ImportError:
+    HAS_PYTHON26 = False
+
+try:
+    from google.cloud import spanner
+    from google.gax.errors import GaxError
+    HAS_GOOGLE_CLOUD_SPANNER = True
+except ImportError as e:
+    HAS_GOOGLE_CLOUD_SPANNER = False
+
+
+def get_spanner_configuration_name(config_name, project_name):
+    config_name = 'projects/%s/instanceConfigs/regional-%s' % (project_name,
+                                                               config_name)
+    return config_name
+
+
+def instance_update(instance):
+    """
+    Call update method on spanner client.
+
+    Note: A ValueError exception is thrown despite the client succeeding.
+    So, we validate the node_count and display_name parameters and then
+    ignore the ValueError exception.
+
+    :param instance: a Spanner instance object
+    :type instance: class `google.cloud.spanner.Instance`
+
+    :returns True on success, raises ValueError on type error.
+    :rtype ``bool``
+    """
+    errmsg = ''
+    if not isinstance(instance.node_count, int):
+        errmsg = 'node_count must be an integer %s (%s)' % (
+            instance.node_count, type(instance.node_count))
+    if instance.display_name and not isinstance(instance.display_name,
+                                                basestring):
+        errmsg = 'display_name must be an string %s (%s)' % (
+            instance.display_name, type(instance.display_name))
+    if errmsg:
+        raise ValueError(errmsg)
+
+    try:
+        instance.update()
+    except ValueError as e:
+        # The ValueError here is the one we 'expect'.
+        pass
+
+    return True
+
+
+def main():
+    module = AnsibleModule(argument_spec=dict(
+        instance_id=dict(type='str', required=True),
+        state=dict(choices=['absent', 'present'], default='present'),
+        database_name=dict(type='str', default=None),
+        configuration=dict(type='str', required=True),
+        node_count=dict(type='int', default=1),
+        display_name=dict(type='str', default=None),
+        service_account_email=dict(),
+        credentials_file=dict(),
+        project_id=dict(), ), )
+
+    if not HAS_PYTHON26:
+        module.fail_json(
+            msg="GCE module requires python's 'ast' module, python v2.6+")
+
+    if not HAS_GOOGLE_CLOUD_SPANNER:
+        module.fail_json(msg="Please install google-cloud-spanner.")
+
+    if not check_min_pkg_version(CLOUD_CLIENT, CLOUD_CLIENT_MINIMUM_VERSION):
+        module.fail_json(msg="Please install %s client version %s" %
+                         (CLOUD_CLIENT, CLOUD_CLIENT_MINIMUM_VERSION))
+
+    mod_params = {}
+    mod_params['state'] = module.params.get('state')
+    mod_params['instance_id'] = module.params.get('instance_id')
+    mod_params['database_name'] = module.params.get('database_name')
+    mod_params['configuration'] = module.params.get('configuration')
+    mod_params['node_count'] = module.params.get('node_count')
```

Comment:
It looks like `node_count` is updated regardless of whether the user specifies it. I think if a user updates to add a table, it might not be their intent to downsize their cluster to a single node (they just want a new table). This is kinda tricky, but it might be best to:

1. If the instance is new, set to 1 by default (or user-provided value)
2. If the instance exists, and the user provides no node_count, do nothing
3. If the instance exists, and the user provides node_count, change cluster size. 

Does that sound reasonable?